frame/staking/src/lib.rs

```diff
@@ -2307,19 +2314,29 @@ pub mod pallet {
 			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 			let stash = ledger.stash;
 
-			// If the caller is not the controller, we want to check that the minimum bond
-			// requirements are not satisfied, and thus we have reason to chill this user.
+			// If the caller is not the controller, we first check that we are near our threshold
+			// limit, and then we also check that the minimum bond requirements are not satisfied,
+			// and thus we are justified to chill this user.
 			//
 			// Otherwise, if caller is the same as the controller, this is just like `chill`.
 			if caller != controller {
 				let min_active_bond = if Nominators::<T>::contains_key(&stash) {
+					let max_nominator_count = MaxNominatorsCount::<T>::get().ok_or(Error::<T>::CannotChillOther)?;
+					let current_nominator_count = CounterForNominators::<T>::get();
+					// Safe for up to hundreds of millions of users.
+					let ratio = current_nominator_count.saturating_mul(100u32) / max_nominator_count.max(1u32);
+					ensure!(ratio > 75, Error::<T>::CannotChillOther);
```

Comment:
We should probably extract the 75 to a governance-controlled parameter. This seems like the sort of thing we'll want to be able to tune without requiring a chain upgrade.